### ISSUE TITLE
Error about backend when trying to run pyfr in DEMO mode

### ISSUE THREAD

**Issue by chetnieter** [Kitware/HPCCloud#518]
I get the following error when I try to run pyfr with the instructions from [simput.md](HPCCloud/docs/usage/simput.md) while using the vagrant VM with DEMO=1.

> [20:24:24.385] ERROR: Exception raise by task.
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/hpccloud/cumulus/cumulus/taskflow/__init__.py", line 117, in wrapped
    return func(celery_task, *args, **kwargs)
  File "/opt/hpccloud/hpccloud/server/taskflows/hpccloud/taskflow/pyfr.py", line 305, in setup_input
    update_config_file(task, client, *args, **kwargs)
  File "/opt/hpccloud/hpccloud/server/taskflows/hpccloud/taskflow/pyfr.py", line 158, in update_config_file
    backend_section = 'backend-%s' % kwargs['backend']['type']
KeyError: 'type'

**Comment by cjh1** [comment]
Looks like the backend type is being passed to the taskflow. Can you look at what the cluster configuration looks like in mongo?

**Comment by chetnieter** [comment]
```javascript
{ "config" : {
		"pyfr" : {
			"openmp" : [
				{
					"cblas" : "/usr/lib/libblas.so",
					"name" : "demo_profile"
				}
			]
		},
		"jobOutputDir" : "/tmp",
		"host" : "vagrant-ubuntu-trusty-64",
		"ssh" : {
			"key" : "57d9...",
			"publicKey" : "ssh-rsa AAAAB3NzaC1y...njz0S2w== cumulus generated access key",
			"user" : "demo",
			"passphrase" : "4TCIY..."
		},
		"scheduler" : {
			"type" : "sge"
		},
		"paraview" : {
			"installDir" : "/opt/paraview"
		}
	}
}
```

**Comment by cjh1** [comment]
So it looks like the client is not sending the backend, property. Can you look in your dev console to confirm that.

**Comment by chetnieter** [comment]
Not sure if this is what I am supposed to be looking for but ...
![console](https://cloud.githubusercontent.com/assets/4336975/18526285/f9241028-7a8d-11e6-9af1-f5375613b27c.png)

**Comment by cjh1** [comment]
It was the payload of the /taskflows/:id/start request I was interested in, should be able to see it in the network view.

**Comment by cjh1** [comment]
@TristanWright This seems to be a UI state issue. If you don't select a backend ( for the first time ), then no backend it sent. If I select the one option in the backend drop-down then it is sent.

**Comment by chetnieter** [comment]
Looks like it is sending an empty backend property.
```javascript 
{
{
  "backend":{},
  "input":{"folder":{"id":"57d961260640fd4ac2d59a3e"},
  "meshFile":{"id":"57d960d30640fd4ac2d59a36"},
  "iniFile":{"id":"57d961290640fd4ac2d59a41"}},
  "output":
    {"folder":     
        {"id":"57d961260640fd4ac2d59a3d"}},
          "numberOfGpusPerNode":0,
          "numberOfSlots":1,
          "queue":"",
          "cluster":{"_id":"57d95e900640fd4ac2d59a2a"}
   }
}
```

**Comment by cjh1** [comment]
Yep, looks like the dropdown needs to be clicked on the trigger it being sent.

**Comment by chetnieter** [comment]
Yeah - I should have mentioned I tried to select an option from the dropdown and it didn't give me any.

**Comment by TristanWright** [comment]
closed with #519
